packages/app/src/composables/useRelevantRun.ts

```diff
@@ -1,20 +1,53 @@
 import { gql, useSubscription } from '@urql/vue'
-import { Debug_RelevantRuns_SubscriptionDocument } from '@packages/app/src/generated/graphql'
+import { Debug_RelevantRuns_SubscriptionDocument, Sidebar_RelevantRuns_SubscriptionDocument } from '@packages/app/src/generated/graphql'
+import { useLoginConnectStore } from '@packages/frontend-shared/src/store/login-connect-store'
+
 import { computed } from 'vue'
 
+/**
+ * Using two different subscriptions with different names in order for urql to treat them separately.
+ *
+ * Currently, this composable is used in two places and can have two subscriptions running at one time.
+ * Without having two separately named subscriptions, urql would not tell the websocket that one of the
+ * subscriptions had ended when the component it was registered in was unmounted.
+ */
 gql`
-  subscription Debug_RelevantRuns_Subscription {
-    relevantRuns {
-      current
-      next
-      commitsAhead
+
+  fragment UseRelevantRun on RelevantRun{
+    current
+    next
+    commitsAhead
+  }
+
+  subscription Debug_RelevantRuns_Subscription($location: RelevantRunLocationEnum!) {
+    relevantRuns(location: $location) {
+      ...UseRelevantRun
+    }
+  }
+
+  subscription Sidebar_RelevantRuns_Subscription($location: RelevantRunLocationEnum!) {
+    relevantRuns(location: $location) {
+      ...UseRelevantRun
     }
   }
 
 `
 
-export function useRelevantRun () {
-  const subscriptionResponse = useSubscription({ query: Debug_RelevantRuns_SubscriptionDocument })
+export function useRelevantRun (location: 'SIDEBAR' | 'DEBUG') {
+  const loginConnectStore = useLoginConnectStore()
+
+  const shouldPause = computed(() => {
+    return !loginConnectStore.project.isProjectConnected
+  })
+
+  //Switch the subscription query depending on where it was registered from
+  let query = Debug_RelevantRuns_SubscriptionDocument
+
+  if (location === 'SIDEBAR') {
+    query = Sidebar_RelevantRuns_SubscriptionDocument
+  }
+
+  const subscriptionResponse = useSubscription({ query, variables: { location }, pause: shouldPause })
```

Comment:
Minor, but do we need to pass a computed property to `pause:`? I kind of expected the state of the `loginConnectStore` to be reactive. 🤔

Reply:
In this case, yes it needs to be wrapped. Or at least I am 99% sure.  If you look at this screenshot, you can see that the root part of the store (`project` in this case) is a `ref`, but the `isProjectConnected` is not.  So if pause was set directly to `!loginConnectStore.project.isProjectConnected` then it would just be passed `true` or `false` and not a `ref`.  I have run into situations where I was wondering why something was not reactively responding to a change and realized I was not passing around a `ref`, but just a raw value.  

<img width="1702" alt="image" src="https://user-images.githubusercontent.com/1702361/215094874-fa1729d9-71bd-4abe-9a32-f41f078667ae.png">